### PR TITLE
Add --cli option

### PR DESCRIPTION
**Pull request by AdityaGarg8**
This allows running this on cli like github actions by eliminating the need to use enter using `--cli` argument.

### CODE REVIEW

**Comment by rzc0d3r** [comment]
Maybe it would be better to make a argument that disables all input('Run anyway?') and etc similar commands?

**Comment by AdityaGarg8** [comment]
Maybe use `--no-input`. Also, I hope the 2 enter commands addressed in this PR are the only inputs?

**Comment by rzc0d3r** [comment]
Yes, from a specific :

https://github.com/rzc0d3r/ESET-KeyGen/blob/main/main.py#L42

https://github.com/rzc0d3r/ESET-KeyGen/blob/main/main.py#L54

Argument disabling this: --force

**Comment by AdityaGarg8** [comment]
Changed

**Comment by rzc0d3r** [comment]
Great, but the error output needs to be left in. How will people copy the Traceback and send it to me? They will have to run the project in the console, not just click on it in explorer.

**Comment by rzc0d3r** [comment]
I have a test version with Traceback output and new JavaScript code lying around on my PC. But I need to test it well.

**Comment by AdityaGarg8** [comment]
My main reason for this PR was for using GitHub actions. Makes it easier and much safer to get licenses without the risk of getting banned.

**Comment by rzc0d3r** [comment]
> Great, but the error output needs to be left in. How will people copy the Traceback and send it to me? They will have to run the project in the console, not just click on it in explorer.

Oh. Well, if they run it from explorer, then --force will not be passed and then the project will run with user input.

**Comment by AdityaGarg8** [comment]
Yes exactly.

**Review by rzc0d3r — CHANGES_REQUESTED** [review]
Don't touch the Press Enter line... at the end of the file

**Comment by AdityaGarg8** [comment]
> Don't touch the Press Enter line... at the end of the file

This shall fail runs on CIs.

**Comment by AdityaGarg8** [comment]
If I remove the requirements of pressing enter, I can generate license using GitHub actions like this:

https://github.com/AdityaGarg8/random/actions/runs/6918918950

**Comment by rzc0d3r** [comment]
Sorry I just don't know how github cli works. Is your commit fully working? If so, I'll merge it.

**Comment by AdityaGarg8** [comment]
Yes, if I remove the need of pressing enter

**Comment by AdityaGarg8** [comment]
If you want, I can help you to get a GitHub CI on your repo to generate licenses

**Comment by rzc0d3r** [comment]
> If you want, I can help you to get a GitHub CI on your repo to generate licenses

What features does it add? I've never encountered this in my life..

**Comment by AdityaGarg8** [comment]
Just like the link I sent you, If you run this manually, it shall use your scripts and generate a license key and display it. Although, I doubt it shall be needed in your repo. I just asked, in case you wanted help.

**Comment by rzc0d3r** [comment]
I understand that through Actions you can run the project in code space and get the results directly on github? Without having to do everything on the main system?

**Comment by AdityaGarg8** [comment]
Yes

**Comment by rzc0d3r** [comment]
It's not a bad feature. Can people who don't have a github account do the same?

**Comment by AdityaGarg8** [comment]
No. The person has to fork your repo or you have generate the license on your repo.

**Comment by rzc0d3r** [comment]
What kind of license?

**Comment by AdityaGarg8** [comment]
I meant the license key

**Comment by rzc0d3r** [comment]
Well, I don't think that's necessary just yet. Idk

**Comment by rzc0d3r** [comment]
Because it looks pretty complicated. And this will need to be described in the documentation

**Comment by AdityaGarg8** [comment]
Fair enough. I hope we can come to some decision on this PR?

**Comment by rzc0d3r** [comment]
> Fair enough. I hope we can come to some decision on this PR?

I could approve. But what's the next step?

**Comment by AdityaGarg8** [comment]
Well, in my case, I'll create a GitHub CI in my private repo, which shall clone your repo, and generate a license key. I'll run the workflow whenever my license expires.

**Comment by AdityaGarg8** [comment]
Something like this https://github.com/AdityaGarg8/random/blob/main/.github/workflows/GenerateChecksum.yml

**Comment by AdityaGarg8** [comment]
Although it needs some cleanup

**Review by rzc0d3r — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -97,4 +100,4 @@ def chrome_driver_installer_menu(): # auto updating or installing chrome driver
             logger.console_log(info.split('Stacktrace:', 1)[0], logger.ERROR)
         else:
             logger.console_log(str(E), logger.ERROR)
-    input('Press Enter...')
\ No newline at end of file
+    input('Press Enter...')
```
Comment (AdityaGarg8) [inline main.py:103]: This needs to be addressed

**Comment by AdityaGarg8** [comment]
Oh it got merged.

**Comment by AdityaGarg8** [comment]
> Oh it got merged.

Now it got reverted?

**Comment by rzc0d3r** [comment]
> > Oh it got merged.
> 
> Now it got reverted?

Lmao. I made the mistake of merging the code. I had to return the return. Don't worry, your code is already in the project.

**Comment by rzc0d3r** [comment]
For cli to work consistently, do you need to remove input('Press Enter...') at the end of main.py?

**Comment by AdityaGarg8** [comment]
Your forgot to address this line https://github.com/rzc0d3r/ESET-KeyGen/blob/main/main.py#L103

**Comment by AdityaGarg8** [comment]
> For cli to work consistently, do you need to remove input('Press Enter...') at the end of main.py?

Yes

**Comment by AdityaGarg8** [comment]
Is there any solid reason to not remove the last line though, maybe I could help there?

**Comment by rzc0d3r** [comment]
I can add the --cli key and then it will disable all inputs.

**Comment by AdityaGarg8** [comment]
Cool

**Comment by AdityaGarg8** [comment]
Also, thanks a lot for this project. Saved me a couple of bucks:)

**Comment by rzc0d3r** [comment]
> Also, thanks a lot for this project. Saved me a couple of bucks:)

https://github.com/rzc0d3r/ESET-KeyGen/commit/e8658a66dac5d36300e85ec11d4ec26109593b3b. You're welcome. Thank you.

**Pull request merged**